DirectProgramming/C++SYCL_FPGA/Tutorials/Features/speculated_iterations/sample.json

```diff
@@ -1,6 +1,6 @@
 {
   "guid": "66A57127-1F8D-4769-8CCB-16ECC56A446F",
-  "name": "Speculated Iterations",
+  "name": "speculated_iterations",
```

Comment:
Do we want to do that? This is the name that would appear in the code samples browser for this sample
1/ Do we want to keep plain English or the actual feature name?
2/ If we decide to make this change, we should make it to all the other samples as well (which I'm ok with as long as we keep them uniform

Reply:
I was following the convention I thought was emerging from several other similar feature samples. I'm okay with whatever you decide.

Reply:
I'd say:
- In the readme, we can use the name of the feature, e.g. `kernel_args_restrict` sample
- In the `samples.json`, use a human readable name as this is what people are going to browse, e.g. "Pointer Aliasing and the Restrict Directive" as this is the name of the doc page: https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2023-0/pointer-aliasing-and-the-restrict-directive.html

What are your thoughts? In that case you would not change the `sample.json` files and I would look for someone to do it

Reply:
I think we should be consistent. If the readme has one name and the sample browsers use another (as determined by the name in the sample.json file) then it's confusing. We understand the difference. Others will not.

Reply:
That makes sense, in that case what would be your recommendation:
1/ use the feature name, e.g. `kernel_args_restrict`
2/ use the doc name, e.g. "Pointer Aliasing and the Restrict Directive"

In any case, do not make the modifications to the sample.json files, I'll make it to all the samples at once.

Reply:
I think we should use option 2, use the doc name. Make them readable to humans who are looking to solve problems. Using the feature name assumes a developer already knows what to look for.